unstructured/ingest/connector/clarifai.py

```diff
@@ -0,0 +1,133 @@
+    
+        self._client = Inputs(app_id = self.connector_config.app_id, 
+                             user_id = self.connector_config.user_id, 
+                             pat = clarifai_pat)
+        return self._client
+        
+        
+    @requires_dependencies(["clarifai"], extras="clarifai")
+    @DestinationConnectionError.wrap
+    def initialize(self):
+        _ = self.client  
+       
+    def check_connection(self):
+        try:
+            _ = self.client
+        except Exception as e:
+            logger.error(f"Failed to validate connection {e}", exc_info=True)
+            raise DestinationConnectionError(f"failed to validate connection: {e}")
+        
+    
+    def normalize_dict(self, element_dict: dict) -> dict:
+        """Modifying schema of the dict inorder to compile with clarifai input formats"""
+        return (
+            {
+                "input_id" : str(uuid.uuid4().hex),
+                "text" : element_dict["text"],
+                "metadata" : {
+                    "url": element_dict["metadata"]["data_source"]["url"],
+                    "date_created": element_dict["metadata"]["data_source"]["date_created"],
+                    "date_modified": element_dict["metadata"]["data_source"]["date_modified"],
+                    "date_processed": element_dict["metadata"]["data_source"]["date_processed"],
+                    "file_directory": element_dict["metadata"]["file_directory"],
+                    "filename": element_dict["metadata"]["filename"],
+                    "filetype": element_dict["metadata"]["filetype"],
+                    "languages": element_dict["metadata"]["languages"],
+                    "last_modified": element_dict["metadata"]["last_modified"],
+                }
+            }
+        )
+        
+    def write_dict(self, *args, elements_dict: t.List[t.Dict[str, t.Any]], **kwargs) -> None:
+        
+        from google.protobuf.struct_pb2 import Struct
+        
+        logger.info(
+            f"writing {len(elements_dict)} objects to destination "
+            f"app {self.connector_config.app_id} "
+        )
+        try:
+            batch = self.write_config.batch_size
+            for idx in range (0, len(elements_dict), batch):
+                batch_dict = elements_dict[idx : batch + idx]
+                meta_list = []
+                for elem in batch_dict :
+                    meta_struct = Struct()
+                    meta_struct.update(elem["metadata"])
+                    meta_list.append(meta_struct)
+                    
+                input_batch = [
+                        self._client.get_text_input(
+                            input_id=batch_dict[i]["input_id"],
+                            raw_text=batch_dict[i]["text"],
+                            metadata=meta_list[i],
+                        )
+                        for i, text in enumerate(batch_dict)
+                    ]
```

Comment:
Since `batch_dict` was iterated once, we don't want to again iterate it in list comprehension. WDYT?



```suggestion
                input_batch = []
                for elem in batch_dict :
                    meta_struct = Struct()
                    meta_struct.update(elem["metadata"])
                    input_batch.append(self._client.get_text_input(
                            input_id=elem["input_id"],
                            raw_text=elem["text"],
                            metadata=meta_struct,)
                        )
                        
```